### DISCUSSION TITLE
Is there a way (or can we add  way) to preprocess the identifier?

### DISCUSSION THREAD

**Discussion by sureshvytla** [antonmedv/expr#702] [Q&A] [answered]
For unfortunate reasons we have hyphens (-) in our identifier names, we would like to preprocess identifiers and replace hyphens.

**Comment by antonmedv** [comment, marked as answer]
You can use `$env` to access variables with `-` dashes inside expression

```js
$env["foo-bar-baz"] > 42
```

**Reply by sureshvytla** [reply]
Thanks, it worked. Is there a way to preprocess and decorate with $env["xxxxx"] for the variables  with `-` dashes?

**Reply by antonmedv** [reply]
Use an expression that detects all variables containing dashes. These could be identified using a regular expression or a specific pattern in your code.

Once identified, you can use the Patch function to replace the variables with the desired format. Specifically, you’ll want to wrap the variable name in $env["..."].
